### ISSUE TITLE
Post processing of generated Wasm failed with rustc 1.70.0-nightly 

### ISSUE THREAD

**Issue by ashutoshvarma** [paritytech/cargo-contract#1058]
**Summary**
Building contracts with `rustc 1.70.0-nightly` failed due to error in post processing step due to error in `parity-wasm` while loading the wasm module.

https://substrate.stackexchange.com/questions/7881/error-loading-of-original-wasm-failed

```
 [2/5] Post processing wasm file
ERROR: Loading of original wasm failed

Caused by:
    0: Loading of wasm module at '/home/oreki/Projects/meta-tx/target/ink/registry/wasm32-unknown-unknown/release/registry.wasm' failed
    1: Unknown opcode 192
```

https://github.com/paritytech/cargo-contract/blob/1d28f4ef3863e1f6a438f59890545eb87e1e863a/crates/build/src/lib.rs#L468-L471

**Reproduce**
Update the nightly to `v1.70.0` and build flipper example with `cargo +nightly contract build`

**Comment by peetzweg** [comment]
Just had the same issue. If you are blocked by this, switch to the `stable` toolchain instead of `nightly`.
Afaik `cargo contract` is not aiming to always work with `nightly` and `stable` should be used from now on.

```
rustup default stable
```

**Comment by ashutoshvarma** [comment]
I'm aware of the workarounds,
1. Switch to `stable` rustc
2. If using `openbrush` (using nightly is required for openbrush still), use nightly v1.69.0 or older.

> Afaik cargo contract is not aiming to always work with nightly and stable should be used from now on.

I understand that with ink! v4 `nightly` toolchain is no longer required and `stable` is recommended but `openbrush` which majority of ink! ecosystem is using still depends on nightly, so this issue might help guys who are likely to struggle with newer nightly toolchain.
Also, I don't think it is a cargo-contract specific problem, it it more an issue in `parity-wasm` which although deprecated, is still used inside cargo-contract. That adds more reason to replace parity-wasm with something that is not deprecated (like wasm-opt/binaryen tools).

**Comment by ganesh1233456** [comment]
I am still facing issue even though I change nightly v1.68 or v1.69 or v1.70 it doesn't work.

**Comment by ashutoshvarma** [comment]
@ganesh1233456 try with nightly `rustc 1.67.1 (d5a82bbd2 2023-02-07)`, it worked for me.

**Comment by blairmunroakusa** [comment]
@ashutoshvarma ... I am having a difficult time figuring out how to install and switch to `rustc 1.67.1 (d5a82bbd2 2023-02-07)`.  Could you please share how you accomplish this?

**Comment by ganesh1233456** [comment]
@blairmunroakusa my system has rust installed already then I use `rustup default <version>` to switch to specific version. But the things is my issue completely not solved, sometimes it works and sometime not work.

**Comment by yahortsaryk** [comment]
> @blairmunroakusa my system has rust installed already then I use `rustup default <version>` to switch to specific version. But the things is my issue completely not solved, sometimes it works and sometime not work.

I have the same issue. I'm compiling my source code (`ink! v4.1.0`) using `rustc 1.67.1` and enabled `+nigthly` flag. Sometimes the build finishes successfully, and sometimes I'm getting `Unknown opcode 192` on the post-processing wasm file step. It really happens randomly. 

Please, let us know how to fix this as it is really annoying issue for CI/CD processes

**Comment by yahortsaryk** [comment]
I'm not sure what's happening on my local environment, I suggest anybody who is struggling with this build problem try to refer the [Installation using Docker Image](https://github.com/paritytech/cargo-contract#installation-using-docker-image) section and compile your contracts using official docker images. It works for me

**Comment by blairmunroakusa** [comment]
Ok @yahortsaryk @ganesh1233456 ...got it figured out on my system:

First:
```
rustup component add rust-src --toolchain nightly-2023-02-07-aarch64-apple-darwin
```
Then:
```
cargo +nightly-2023-02-07 contract build
```
Still running updated rustc as default:
![image](https://user-images.githubusercontent.com/69293813/230799234-12a4e140-9323-4617-8f46-1dc59b6302a0.png)

Thank you @ashutoshvarma

**Comment by AgentPoles** [comment]
Thanks a lots, worked for me! Just to mention for anyone trying this method, if you have "rlib" crate-types in your project, you might wanna skip those and apply to actual contracts: "cdylib" crate-types

**Comment by blairmunroakusa** [comment]
Now I'm having same opcode throw when trying to run substrate-contracts-node.

**Comment by LusterNetwork** [comment]
It worked and build my contract but unable to deploy it on blockchain

**Comment by upendra-eth** [comment]
> Ok @yahortsaryk @ganesh1233456 ...got it figured out on my system:
> 
> First:
> 
> ```
> rustup install nightly-2023-02-07-aarch64-apple-darwin
> ```
> 
> ```
> rustup component add rust-src --toolchain nightly-2023-02-07-aarch64-apple-darwin
> ```
> 
> Then:
> 
> ```
> cargo +nightly-2023-02-07 contract build
> ```
> 
> Still running updated nightly rustc as default: ![image](https://user-images.githubusercontent.com/69293813/230799234-12a4e140-9323-4617-8f46-1dc59b6302a0.png)
> 
> Thank you @ashutoshvarma

thanks, it worked but it's not  deploying on blockchain

**Comment by ascjones** [comment]
Rust `1.70` is producing the `sign-ext` opcode which is neither compatible with `parity-wasm` which is what produces the error in `cargo-contract`, nor in[ `pallet-contracts` itself](https://github.com/paritytech/substrate/blob/master/frame/contracts/src/wasm/mod.rs#L218).

Ideally we would disable that feature but it appears that this does not work, see this upstream issue: https://github.com/rust-lang/rust/issues/109807.

I will close this issue as a duplicate of https://github.com/paritytech/cargo-contract/issues/1139, where we can continue any discussion for the resolution of this.
